### ISSUE TITLE
Feature: Layout/SpaceInsideArrayLiteralBrackets should allow comments inside empty lines.

### ISSUE THREAD

**Issue by ayacai115** [rubocop-hq/rubocop#7512]
follow-up: https://github.com/rubocop-hq/rubocop/pull/7511#issuecomment-554840258
## Problem
After this PR(https://github.com/rubocop-hq/rubocop/pull/7511 ), comments between empty lines would be destroyed on auto-correct. 

```rb
# before
foo = [
# Intentionally left empty
]

# auto-corrected
foo = []
```

## Possible solution
Fix `Layout/SpaceInsideArrayLiteralBrackets` to allow comments inside empty lines.
```rb
# before
foo = [
# Intentionally left empty
]

# auto-corrected
foo = [
# Intentionally left empty
]
```

Let me know if you come up with any other alternatives😃

**Comment by buehmann** [comment]
Have you actually tried this? Comments are not touched in this case.

**Comment by Drenmi** [comment]
> Have you actually tried this? Comments are not touched in this case.

He might have taken my word for it, and to be honest: I didn't try it. 😅 

There are a few options as I can see. Listed in order of personal preference. 🙂 

1. Provide a configuration option, e.g. `PreserveComments` to let the user decide.
2. Don't auto-correct (but register an offense) when it would destroy comments.
3. Try to be smart and hoist the comment. (From experience very error prone.)

I think a few other cops would have the same implication, like `Style/EmptyMethod` for example. We could check what those cops do. Ideally we should keep it consistent.

**Comment by ayacai115** [comment]
>Have you actually tried this? Comments are not touched in this case.

I just tried and you're right:sweat_smile:
The comment wasn't destroyed.

**Comment by ayacai115** [comment]
So this issue can be closed now?

**Comment by ayacai115** [comment]
I'm gonna close this issue because `Layout/SpaceInsideArrayLiteralBrackets` won't cause the problem.

@Drenmi 
>I think a few other cops would have the same implication, like Style/EmptyMethod for example. We could check what those cops do. Ideally we should keep it consistent.

We can discuss that in another issue, so could you open a new one if you like?😃
